js/src/mvtUtils.ts

```diff
@@ -1,51 +1,34 @@
 import { VectorTile } from '@mapbox/vector-tile';
 import Protobuf from 'pbf';
 
-export interface MVTFeature {
-    id: number;
-    geometry: { x: number; y: number }[][];
-    properties: Map<string, unknown>;
-}
-
-export interface MVTLayer {
-    name: string;
-    features: MVTFeature[];
-}
-
-export function parseMvtTile(mvtTile: Buffer): MVTLayer[] {
+export function parseMvtTile(mvtTile: Buffer): any {
     const vectorTile = new VectorTile(new Protobuf(mvtTile));
-
     const layers = [];
     for (const layerName of Object.keys(vectorTile.layers)) {
         const layer = vectorTile.layers[layerName];
         const features = [];
         layers.push({ name: layerName, features });
         for (let i = 0; i < layer.length; i++) {
-            const feature = layer.feature(i);
-            const id = feature.id;
-            const geometry = feature.loadGeometry();
-
-            const properties = new Map<string, unknown>();
-            for (const [name, value] of Object.entries(feature.properties)) {
-                properties.set(name, value);
-            }
-
-            features.push({ id, geometry, properties });
+            features.push(layer.feature(i));
         }
     }
-
-    return layers;
+    return {
+        layers: layers
+    }
 }
 
-export function parseMvtTileFast(mvtTile: Buffer): MVTLayer[] {
+export function parseMvtTileFull(mvtTile: Buffer): any {
     const vectorTile = new VectorTile(new Protobuf(mvtTile));
-
+    const layers = [];
     for (const layerName of Object.keys(vectorTile.layers)) {
         const layer = vectorTile.layers[layerName];
+        const features = [];
+        layers.push({ name: layerName, features });
         for (let i = 0; i < layer.length; i++) {
-            layer.feature(i).loadGeometry();
+            const feature = layer.feature(i);
+            feature.geom = feature.loadGeometry();
```

Comment:
I'm getting an error here:

```bash
src/mvtUtils.ts:29:21 - error TS2339: Property 'geom' does not exist on type 'VectorTileFeature'.

29             feature.geom = feature.loadGeometry();
                       ~~~~


Found 1 error in src/mvtUtils.ts:29
```

Reply:
Fixed in [daf1474](https://github.com/maplibre/maplibre-tile-spec/pull/162/commits/daf147422eefc43d2378c35bb125f5790b48a7b2)